test/main_unit.C

```diff
@@ -153,10 +153,14 @@ int main(void)
 
   long int divl = 1;
   long long int divll = 1;
+  unsigned long int udivl = 1;
+  unsigned long long int udivll = 1;
```

Comment:
Could we add some tests specifically using size_t and ptrdiff_t here?  At least that way if my vague "some platforms equate integer types that others don't" recollection does cause problems, we'll see that crop up first in unit tests rather than user code.